src/models/Order.js

```diff
@@ -115,6 +117,30 @@ OrderSchema.pre('findOne', async function (next) {
   console.log('Pre-findOne finished');
   next();
 });
+
+// Middleware to send order completion email when the order status changes to "complete"
+OrderSchema.post('save', async function (doc) {
+  try {
+    if (doc.orderStatus === 'complete') {
+      const user = await User.findById(doc.user); // Assuming you have a User model
+
+      const orderItemsWithFullAlbum = await doc
+        .populate({ path: 'orderItems.album' })
+        .execPopulate();
```

Comment:
`execPopulate` is not a function. It was removed in mongoose v6 (we are using v7): [https://github.com/Automattic/mongoose/blob/f552fe286a42373e50970072352b66a1fd443a1c/docs/migrating_to_6.md#removed-execpopulate](https://github.com/Automattic/mongoose/blob/f552fe286a42373e50970072352b66a1fd443a1c/docs/migrating_to_6.md#removed-execpopulate)

We can populate by calling `await` + `populate(...)` and we can chain populate calls by using an array. This allows us to populate both the full user object and the full orderItems.album objects:

```javascript
const orderItemsWithFullAlbum = await doc.populate(['user', { path: 'orderItems.album' }])
```